src/yarp.c

```diff
@@ -4216,6 +4216,33 @@ next_newline(const char *cursor, ptrdiff_t length) {
     return memchr(cursor, '\n', (size_t) length);
 }
 
+// Return the length of the line ending string starting at +cursor+, or 0 if it is not a line
+// ending. This function is intended to be CRLF/LF agnostic.
+static inline size_t
+match_line_ending_addr(yp_parser_t *parser, const char *cursor) {
+    if (peek_addr(parser, cursor) == '\n') {
+        return 1;
+    }
+    if (peek_addr(parser, cursor) == '\r' && peek_addr(parser, cursor + 1) == '\n') {
+        return 2;
```

Comment:
It seems like when we use this always use it to find the location of the `\n`. Do you think we can return that `char *` location instead, or maybe there are cases I missed?

Reply:
:thinking: Let me give it a tr.

Reply:
There are a few places where the code branches based on whether we matched `\r` or `\r\n`, and those situations become significantly messier if we return a pointer.

Reply:
Makes sense! thanks for looking into it!

Reply:
Hmm, maybe you're right, I'll throw that work on a branch so we can compare.

Reply:
This ended up requiring code that was (subjectively) harder to reason about, and I actually had trouble debugging it. Offsets seem to fit the code patterns better, so I'm going to stick with that.